### DISCUSSION TITLE
Is it possible to build a host dependency in release version only?

### DISCUSSION THREAD

**Discussion by tsondergaard** [microsoft/vcpkg#37760] [General]
qt5-base depends on qt5-base in a host version ([link](https://github.com/microsoft/vcpkg/blob/master/ports/qt5-base/vcpkg.json#L26)) and this host version is built in both a debug and release configuration. The host debug version takes both many minutes of build time and quite a bit of disk space (1.5 GiB).

I expect the host dependency on qt5-base is to provide tools like moc and uic, but I imagine it always uses the release build of these. At least it should. Is there a way to specify that a host dependency is only needed as release build?

**Comment by Neumann-A** [comment]
`--host-triplet x64-windows-release` also just setting the target as the host also helps with this (if you are on windows)

**Comment by tsondergaard** [comment]
@Neumann-A, thanks that's great. Shouldn't x64-windows-release be the default host triplet on Windows?

**Comment by tsondergaard** [comment]
> also just setting the target as the host also helps with this (if you are on windows)?

What does this actually mean? If I am building on the host should I set the target and host triplets to the same? Will that prevent building qt5-base twice? It seems logical that it will. Let me try that.

**Reply by Neumann-A** [reply]
it is enough to set the --host-triplet and leave the --triplet undefined.  (If you are using cli. cmake maybe has to define both)

**Comment by tsondergaard** [comment]
It works. Thanks @Neumann-A, my vcpkg_installed folder on linux has shrunk from 11 GiB to 8.1 GiB. What I did was set VCPKG_HOST_TRIPLET and VCPKG_TARGET_TRIPLET to the same in my CMakePresets.json file.
